pkg/filters/hls.go

```diff
@@ -55,16 +55,81 @@ func (h *HLSFilter) FilterManifest(filters *parsers.MediaFilters) (string, error
 	return filteredManifest.String(), nil
 }
 
+// Returns true if specified variant passes all filters
 func (h *HLSFilter) validateVariants(filters *parsers.MediaFilters, v *m3u8.Variant) bool {
 	if filters.DefinesBitrateFilter() {
 		if !(h.validateBandwidthVariant(filters.MinBitrate, filters.MaxBitrate, v)) {
 			return false
 		}
 	}
 
+	variantCodecs := strings.Split(v.Codecs, ",")
+
+	if filters.Audios != nil {
+		supportedAudioTypes := map[string]struct{}{}
+		for _, at := range filters.Audios {
+			supportedAudioTypes[string(at)] = struct{}{}
+		}
+		res := validateVariantCodecs(audioContentType, variantCodecs, supportedAudioTypes)
+		if !res {
+			return false
+		}
+	}
+
+	if filters.Videos != nil {
+		supportedVideoTypes := map[string]struct{}{}
+		for _, vt := range filters.Videos {
+			supportedVideoTypes[string(vt)] = struct{}{}
+		}
+		res := validateVariantCodecs(videoContentType, variantCodecs, supportedVideoTypes)
+		if !res {
+			return false
+		}
+	}
+
+	if filters.CaptionTypes != nil {
+		supportedCaptionTypes := map[string]struct{}{}
+		for _, ct := range filters.CaptionTypes {
+			supportedCaptionTypes[string(ct)] = struct{}{}
+		}
+		res := validateVariantCodecs(captionContentType, variantCodecs, supportedCaptionTypes)
+		if !res {
+			return false
+		}
+	}
+
 	return true
 }
 
+// Returns true if the given variant (variantCodecs) should be allowed through the filter for supportedCodecs of filterType
+func validateVariantCodecs(filterType ContentType, variantCodecs []string, supportedCodecs map[string]struct{}) bool {
+	var matchFilterType func(string) bool
+	typeInVariant := 0
+	matchInVariant := 0
```

Comment:
because ints have a default value of 0, I think you can replace this with:

```suggestion
var typeInVariant, matchInVariant int